api/resources_portal/views/material_request_issue.py

```diff
@@ -0,0 +1,209 @@
+
+
+class IsRequesterOrIsInOrg(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return (
+            request.user == obj.material_request.requester
+            or request.user in obj.material_request.material.organization.members.all()
+        )
+
+
+# def send_material_request_issue_notif(notif_type, issue, notified_user):
+#     notification = Notification(
+#         notification_type=notif_type,
+#         notified_user=notified_user,
+#         associated_user=issue.assigned_to,
+#         associated_material=issue.request.material,
+#         associated_organization=issue.request.material.organization,
+#     )
+#     notification.save()
+
+
+# def send_transfer_update_notif(status, request):
+#     if status == "OPENED":
+#         send_material_request_issue_notif("REQUEST_ISSUE_OPENED", request, request.requester)
+#     elif status == "CLOSED":
+#         send_material_request_issue_notif("REQUEST_ISSUE_CLOSED", request, request.requester)
+#     else:
+#         return
+
+
+class MaterialRequestIssueViewSet(viewsets.ModelViewSet):
+    # No delete, these just get closed.
+    http_method_names = ["get", "list", "post", "put", "patch", "head", "options"]
+
+    def get_queryset(self):
+        if self.action == "list":
+            requests_made_by_user = MaterialRequest.objects.filter(requester=self.request.user)
+            organizations = get_objects_for_user(
+                self.request.user, "view_requests", klass=Organization.objects
+            )
+            requests_viewable_by_user = MaterialRequest.objects.filter(
+                material__organization__in=organizations
+            )
+
+            viewable_requests = requests_made_by_user.union(requests_viewable_by_user)
+            return MaterialRequestIssue.objects.filter(
+                material_request_id__in=viewable_requests.values("id")
+            )
+            # viewable_requests = requests_made_by_user.union(requests_viewable_by_user)
+            # return MaterialRequestIssue.objects.filter(material_request__in=viewable_requests)
+        else:
+            return MaterialRequestIssue.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MaterialRequestIssueSerializer
+
+        return MaterialRequestIssueDetailSerializer
+
+    def get_permissions(self):
+        if self.action == "create":
+            permission_classes = [IsAuthenticated, IsRequester]
+        if (
+            self.action == "list"
+            or self.action == "retrieve"
+            or self.action == "update"
+            or self.action == "partial-update"
+        ):
+            permission_classes = [IsAuthenticated, IsRequesterOrIsInOrg]
+        else:
+            permission_classes = [IsAuthenticated, IsAdminUser]
+
+        return [permission() for permission in permission_classes]
+
+    def create(self, request, *args, **kwargs):
+        # TODO: Generate a notification
```

Comment:
It looks like there are some notifications. Is this TODO TODONE?

Reply:
It is!